### ISSUE TITLE
WCPay Subscriptions compatibility with All Products for WooCommerce Subscriptions

### ISSUE THREAD

**Issue by aprea** [Automattic/woocommerce-payments#3524]
### Description
Determine the level of compatibility between WCPay Subscriptions and [All Products for WooCommerce Subscriptions](https://woocommerce.com/products/all-products-for-woocommerce-subscriptions/). Then, either fix any compatibility issues (if straightforward), or work with other teams to get fixes prioritised (if more complex).

product | maintainer | status | comments
--- | --- | --- | --- 
[All Products for WooCommerce Subscriptions](https://woocommerce.com/products/all-products-for-woocommerce-subscriptions/) | @Automattic/somewherewarm  | 🟡 | WC APFS checks for the main plugin class `WC_Subscriptions` before loading. An admin notice is displayed.

Related issues:
- https://github.com/Automattic/woocommerce-payments/issues/3523
- https://github.com/Automattic/woocommerce-payments/issues/3525
- https://github.com/Automattic/woocommerce-payments/issues/3526

**Note:** Before working on this issue, please check whether any of the above issues are started/completed and work with the assignee to discuss the best approach for these compatibility issues.

### Acceptance criteria

* Determine how the extension builds on subscriptions - what functionality it adds, and if there are any differences in WCPay Subscriptions.
* Test and log issues for any fixes needed.
* Implement PRs for those fixes (if straightforward), or work with other teams to get fixes prioritised (if more complex).
* Note down any issues or opportunities with extensibility APIs. E.g. if a hook needs to be added or changed for a compatibility fix.

From there, we can build on this to provide guidance for third-party extensions integrating with Subscriptions. For example, documentation and or sample code answering questions like these:

* What behaviour can be extended and any differences between WCPay Subscriptions and WC Subscriptions.
* How to detect subscriptions functionality and initialise support code - best practice recommendations.
* How to test an extension alongside subscriptions - e.g. what flows might break.

### Additional context
Split from https://github.com/Automattic/woocommerce-payments/issues/2816

P2 pdjTHR-fI-p2

**Comment by jimjasson** [comment]
Hey @aprea 👋🏻

Great initiative! 

I wanted to let you know that we have already ran a series of tests between All Products for WooCommerce Subscriptions and WooCommerce Payments and we have identified the following issues: 
- content/plan switching is not supported, 
- adding products to existing Subscriptions is not supported and; 
- the subscriptions template directory has changed location. 

We have already created a PR that resolves these issues on our side: https://github.com/woocommerce/woocommerce-all-products-for-subscriptions/pull/413 which is close to being merged. 

So, if you jump into any of the above issues, feel free to grab the latest APFS dev version!
